src/compiler/types.rs

```diff
@@ -1389,6 +1495,12 @@ pub struct DiagnosticMessage {
     pub message: &'static str,
 }
 
+#[derive(Clone, Debug)]
+pub struct DiagnosticMessageChain {
+    pub message_text: String,
+    pub next: Option<Vec<DiagnosticMessageChain>>,
```

Comment:
This is probably mimicking the TS code, but seems weird to have an option of a vec (which can be empty), unless empty vec and `None` mean different things.